Sources/Web3Inbox/PushClientProxy/PushClientProxy.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+final class PushClientProxy {
+
+    private let client: WalletPushClient
+
+    var onSign: SigningCallback
+    var onResponse: ((RPCResponse) async throws -> Void)?
+
+    init(client: WalletPushClient, onSign: @escaping SigningCallback) {
+        self.client = client
+        self.onSign = onSign
+    }
+
+    func request(_ request: RPCRequest) async throws {
+        guard let event = PushWebViewEvent(rawValue: request.method)
+        else { throw Errors.unregisteredMethod }
+
+        switch event {
+        case .approve:
+            let params = try parse(ApproveRequest.self, params: request.params)
+            try await client.reject(id: params.id)
```

Comment:
Just wondering if this is calling the correct method on the `client`? I would expect `.approve(...)` instead

Reply:
sound like something is not right xd